corporate/views/remote_upgrade_page.py

```diff
@@ -0,0 +1,57 @@
+import logging
+
+from django.http import HttpRequest, HttpResponse, HttpResponseRedirect
+from django.shortcuts import render
+from django.utils.translation import gettext as _
+
+from corporate.lib.remote_billing_util import get_identity_dict_from_session
+from corporate.lib.stripe import InitialUpgradeRequest, RemoteRealmBillingSession
+from corporate.models import CustomerPlan
+from zerver.decorator import self_hosting_management_endpoint
+from zerver.lib.exceptions import JsonableError
+from zerver.lib.request import REQ
+from zerver.lib.validator import check_bool
+from zilencer.models import RemoteRealm, RemoteZulipServer
+
+billing_logger = logging.getLogger("corporate.stripe")
+
+
+@self_hosting_management_endpoint
+def remote_realm_upgrade_page(
+    request: HttpRequest,
+    *,
+    realm_uuid: str,
+    manual_license_management: bool = REQ(default=False, json_validator=check_bool),
```

Comment:
I think this `REQ` won't do anything without `has_request_variables`. But also I think we should probably continue using `@typed_endpoint` for the new billing views just to save a migration.

Reply:
Done.